anonymizer/base.py

```diff
@@ -39,18 +40,18 @@ def _prep_init(self, field):
         field_vals = set(x[0] for x in field.model._default_manager.values_list(field.name))
         self.init_values[field] = field_vals
 
-    def get_allowed_value(self, source, field):
+    def get_allowed_value(self, source, field, force_unique=False):
         retval = source()
-        if field is None or retval == '':
+        if field is None:
             return retval
 
         # Enforce unique.  Ensure we don't set the same values, as either
         # any of the existing values, or any of the new ones we make up.
-        unique = getattr(field, 'unique', None)
+        unique = getattr(field, 'unique', None) or force_unique
         if unique:
             self._prep_init(field)
             used = self.init_values[field]
```

Comment:
how big does this `used` set get during anonymization? Is it actually keeping around all the values in the fields across all models?